lib/rmt/gpg.rb

```diff
@@ -0,0 +1,51 @@
+require 'English'
+require 'fileutils'
+require 'tmpdir'
+
+class RMT::Gpg
+  class RMT::Gpg::Exception < RuntimeError
+  end
+
+  def initialize(metadata_file:, key_file:, signature_file:, logger:)
+    @metadata_file = metadata_file
+    @key_file = key_file
+    @signature_file = signature_file
+    @logger = logger
+  end
+
+  def verify_signature
+    @tmpdir = Dir.mktmpdir('rmt-mirror-gpg')
+    @keyring = File.join(@tmpdir, 'keyring')
+
+    run_import_key
+    run_verify_signature
+
+    true
+  ensure
+    FileUtils.rm_rf(@tmpdir)
+  end
+
+  protected
```

Comment:
why `protected` and not `private` ?

Reply:
Why not? In Ruby there's very little difference between `protected` and `private`. I personally prefer `protected`, mostly due to how `protected` works in other languages.

Reply:
exactly, so the thing is that we don't _need_ protected in this case. but now I check other places in the project where we also use `protected` so I guess it's OK :slightly_smiling_face: